### PR TITLE
#30 Github App authentication

### PR DESCRIPTION
**Pull request by henrybley**
Fixed the issues with the github app key so that it is actually authenticating using the generated token

currently it needs both the App id and the installation id would hope in future we only need the App id and the installation id can be queried for. have created a backlog item for this

### CODE REVIEW

File: app/src/test/java/be/xplore/githubmetrics/app/IntegrationTest.java [hunk 1]
```diff
@@ -29,15 +30,21 @@
         webEnvironment = RANDOM_PORT
 )
 @AutoConfigureObservability
-public class JobCountsIntegrationTest {
-
+class IntegrationTest {
```
Comment (tomellm) [inline app/src/test/java/be/xplore/githubmetrics/app/IntegrationTest.java:33]: since there are two other comments might aswell change this to plural xD

**Review by tomellm — APPROVED** [review]
just remove the TestConfig class otherwise nice

File: app/src/test/java/be/xplore/githubmetrics/app/IntegrationTest.java (hunk 1 above)
Comment (henrybley) [reply app/src/test/java/be/xplore/githubmetrics/app/IntegrationTest.java]: done

File: app/src/test/java/be/xplore/githubmetrics/app/TestConfig.java [hunk 2]
```diff
@@ -0,0 +1,8 @@
+package be.xplore.githubmetrics.app;
+
+import org.springframework.stereotype.Component;
+
+@Component
+public record TestConfig() {
+
+}
```
[the remark this answers is not in the document]
Comment (henrybley) [reply app/src/test/java/be/xplore/githubmetrics/app/TestConfig.java]: done

File: app/src/test/java/be/xplore/githubmetrics/app/IntegrationTest.java [hunk 3]
```diff
@@ -73,4 +80,15 @@ void retrieveAndExportShouldCorrectlyDisplayOnActuatorEndpoint() throws Exceptio
                 content().string(Matchers.containsString("workflow_run_jobs{conclusion=\"SUCCESS\",status=\"DONE\",} 2.0"))
         );
     }
+
+    @Test
+    void retrieveAndExportShouldCorrectlyDisplayOnActuatorEndpoint() throws Exception {
```
[the remark this answers is not in the document]
Comment (henrybley) [reply app/src/test/java/be/xplore/githubmetrics/app/IntegrationTest.java]: done

**Pull request merged**